### PR TITLE
Add ability to read cipher key from Secret Manager

### PR DESCRIPTION
**Pull request by bjornandre**
This PR supports the Secret Manager pattern from the [Berglas environment variable reference syntax](https://github.com/GoogleCloudPlatform/berglas/blob/main/doc/reference-syntax.md).

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -66,3 +68,53 @@ def encrypt(request: EncryptionRequest):
...
+        raise Exception(log_msg)
+
+    splitted = without_prefix.split("/", 1)
+
+    if splitted[1] == "":
+        log_msg = f"No name defined in {env_var_value}"
+        logging.error(log_msg)
+        raise Exception(log_msg)
+
+    if "#" in splitted[1]:
```
Comment (mmwinther) [inline app/main.py:117]: This is slightly cryptic, could have a short comment on what we're checking here

File: tests/test_api.py [hunk 2]
```diff
@@ -31,3 +31,16 @@ def test_encrypt():
     assert response.json() == {
         "message": 'dQB+Eaf751En6/j4TQtrcg=='
     }
+
+
+def test_parsing():
```
Comment (mmwinther) [inline tests/test_api.py:36]: This looks like two tests to me

File: app/main.py [hunk 3]
```diff
@@ -66,3 +68,53 @@ def encrypt(request: EncryptionRequest):
 def app_startup():
     """Does some initial startup stuff"""
     logger.info(f"Starting Statbank Authenticator version {__version__} ...")
+    if 'CIPHER_KEY' not in os.environ:
+        raise EnvironmentError('Missing environment variable: CIPHER_KEY')
+
+    if os.environ['CIPHER_KEY'].startswith(SECRET_MANAGER_PREFIX):
```
Comment (mmwinther) [inline app/main.py:74]: Should we not raise an EnvironmentError if this check fails? Or does that mean we've already decoded the key? In that case, would be good to have a comment.

File: app/main.py [hunk 4]
```diff
@@ -66,3 +68,53 @@ def encrypt(request: EncryptionRequest):
...
+        name = f"projects/{project_id}/secrets/{secret_id}/versions/{version}"
+
+        # Access the secret version.
+        response = client.access_secret_version(request={"name": name})
+
+        # Verify payload checksum.
+        crc32c = google_crc32c.Checksum()
+        crc32c.update(response.payload.data)
+        if response.payload.data_crc32c != int(crc32c.hexdigest(), 16):
+            logger.error("Data corruption detected.", response)
```
Comment (mmwinther) [inline app/main.py]: Shouldn't we raise an exception in this case as well? Otherwise we'll just continue on

**Review by mmwinther — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 3 above)
Comment (mmwinther) [inline app/main.py:74]: I think it's worth writing a test for this code block, looks like it's doable by mocking `access_secret_version`

File: app/main.py (hunk 1 above)
Comment (bjornandre) [reply app/main.py:111]: Fixed

File: app/main.py (hunk 4 above)
Comment (bjornandre) [reply app/main.py]: Fixed

File: app/main.py (hunk 3 above)
Comment (bjornandre) [reply app/main.py]: Fixed

Comment (bjornandre) [reply app/main.py]: The cipher key can either be stored directly in the environment variable, or via Secret Manager. Added a comment.

File: tests/test_api.py [hunk 5]
```diff
@@ -4,9 +4,16 @@
...
+from app.main import app, get_project_and_name, get_sm_client
+from google.cloud.secretmanager import SecretManagerServiceClient, AccessSecretVersionResponse, SecretPayload
+from google.auth import credentials
+
+CIPHER_KEY_16_DIGIT = '@NcRfUjXn2r5u8x/'
 
 client = TestClient(app)
+# Local instance of Secret Manager Service that can be used to fake reponses
+sm_client = SecretManagerServiceClient(credentials=credentials.AnonymousCredentials)
+app.dependency_overrides[get_sm_client] = lambda: sm_client
```
Comment (mmwinther) [inline tests/test_api.py:16]: This was clever 👍

**Review by mmwinther — APPROVED** [review]
Great structure and tests 👍

**Pull request merged**